### PR TITLE
Enable OT tracing in Ray

### PR DESCRIPTION
**Pull request by akihikokuroda**
### Summary
This is the first part of the #841. 

### Details and comments
This just enables OT tracing in the Ray components.

The propagation from Ray span to Program span and the OT tracing in the Gateway (django) are still under investigation.

### CODE REVIEW

File: client/quantum_serverless/core/tracing.py [hunk 1]
```diff
@@ -114,3 +114,17 @@ def _trace_env_vars(env_vars: dict, location: Optional[str] = None):
...
+
+    Passed as an argument at Ray start
+    """
+    resource = Resource(attributes={SERVICE_NAME: "Quantum-Serverless: Ray"})
+    provider = TracerProvider(resource=resource)
+    otel_exporter = BatchSpanProcessor(
+        OTLPSpanExporter(endpoint="http://otel-collector:4317/", insecure=True)
+    )
+    provider.add_span_processor(otel_exporter)
+    trace._set_tracer_provider(provider, log=False)
```
Comment (psschwei) [inline client/quantum_serverless/core/tracing.py]: think the lint warning can be disabled via something like  `# pylint: disable=W0212` , but should we? not sure what the python best practices are in this space...

Comment (psschwei) [reply client/quantum_serverless/core/tracing.py]: looks like you fixed it while I was typing that :laughing:

File: client/quantum_serverless/core/tracing.py [hunk 2]
```diff
@@ -114,3 +114,17 @@ def _trace_env_vars(env_vars: dict, location: Optional[str] = None):
...
+
+def setup_tracing() -> None:
+    """Setup Tracing for Ray cluster
+
+    Passed as an argument at Ray start
+    """
+    resource = Resource(attributes={SERVICE_NAME: "Quantum-Serverless: Ray"})
+    provider = TracerProvider(resource=resource)
+    otel_exporter = BatchSpanProcessor(
+        OTLPSpanExporter(endpoint="http://otel-collector:4317/", insecure=True)
```
Comment (psschwei) [inline client/quantum_serverless/core/tracing.py:127]: out of curiosity, would it cause any issues here if an alternative agent host/port is passed in L77 ? I assume those are separate so there's no reason to worry, but figure it doesn't hurt to ask

Comment (akihikokuroda) [reply client/quantum_serverless/core/tracing.py:127]: oops,  it's for my debugging.  I'll change it to
```
f"{agent_host}:{agent_port}"
```
Thanks!

**Review by psschwei — APPROVED** [review]

**Comment by Tansito** [comment]
Oh, true, I don't know why. Maybe a connectivity problem (sometimes happens).

File: client/quantum_serverless/core/tracing.py [hunk 3]
```diff
@@ -74,7 +74,7 @@ def get_tracer(
     provider = TracerProvider(resource=resource)
     if agent_host is not None and agent_port is not None:
         otel_exporter = BatchSpanProcessor(
-            OTLPSpanExporter(endpoint=f"{agent_host}:{agent_port}")
+            OTLPSpanExporter(endpoint=f"{agent_host}:{agent_port}", insecure=True)
```
Comment (Tansito) [inline client/quantum_serverless/core/tracing.py:77]: Is it possible to configure the `insecure` variable through a environment variable?

Maybe I'm wrong but I think we can have some warnings from the security team. What do you think @psschwei @akihikokuroda ? I suppose that can be `True` for development and `False` for production.

Comment (psschwei) [reply client/quantum_serverless/core/tracing.py:77]: Yeah, that should be doable as an envvar, similar to what we do for `SERVICE_NAME` just above

**Review by psschwei — APPROVED** [review]

**Review by IceKhan13 — APPROVED** [review]

**Pull request merged**